### PR TITLE
feat(remote): advanced configurable CSRF token fetching for cds@7.5

### PR DESCRIPTION
**Pull request by arleytm**
Introduce a new configuration option for CSRF token retrieval, allowing users to specify the HTTP method and the URL. This change addresses interoperability issues with some remote services that may require a custom CSRF token handling approach.

answers: https://answers.sap.com/questions/13995293/sap-cap-csrf-token-not-working-with-api-provider-a.html

### CODE REVIEW

File: node.js/remote-services.md [hunk 1]
```diff
@@ -48,6 +50,39 @@ If the remote system you want to consume requires it, you can enable the new CSR
...
+            "kind": "odata",
+            "model": "srv/external/API_BUSINESS_PARTNER",
+            "csrf": {
+              "method": "get",
+              "url": "..."
+            },
+            "csrfInBatch": {
+              "method": "get",
+              "url": "..."
+            }
```
Comment (arleytm) [inline node.js/remote-services.md:73]:
```suggestion
            }
```

**Review by hm23 — APPROVED** [review]

File: node.js/remote-services.md [hunk 2]
```diff
@@ -48,6 +50,35 @@ If the remote system you want to consume requires it, you can enable the new CSR
...
+#### Advanced Configuration
+
+For more advanced scenarios, you can further customize the CSRF-token handling with additional parameters:
+
+```json
+"cds": {
+    "requires": {
+        "API_BUSINESS_PARTNER": {
+            "kind": "odata",
+            "model": "srv/external/API_BUSINESS_PARTNER",
```
Comment (renejeglinsky) [inline node.js/remote-services.md:65]:
```suggestion
            [...]
```

File: node.js/remote-services.md [hunk 3]
```diff
@@ -48,6 +50,35 @@ If the remote system you want to consume requires it, you can enable the new CSR
...
+            "csrf": {
+              "method": "get",
+              "url": "..."
```
Comment (renejeglinsky) [inline node.js/remote-services.md:68]:
```suggestion
            "csrf": {  // [!code focus]
              "method": "get",  // [!code focus]
              "url": "..."  // [!code focus]
```

File: node.js/remote-services.md [hunk 4]
```diff
@@ -48,6 +50,35 @@ If the remote system you want to consume requires it, you can enable the new CSR
     }
 }
 ```
+
+In this example, CSRF handling is enabled for the `API_BUSINESS_PARTNER` service, for regular requests (`csrf: true`) and requests made within batch operations (`csrfInBatch: true`).
+
+#### Advanced Configuration
+
+For more advanced scenarios, you can further customize the CSRF-token handling with additional parameters:
```
Comment (renejeglinsky) [inline node.js/remote-services.md:58]:
```suggestion
Actually `csrf: true` is a convenient preset. If needed, you can further customize the CSRF-token handling with additional parameters:
```

File: node.js/remote-services.md [hunk 5]
```diff
@@ -48,6 +50,34 @@ If the remote system you want to consume requires it, you can enable the new CSR
...
+    }
+}
+```
+
+Here, the CSRF-token handling is customized at a more granular level:
+
+ - `method`: The HTTP method for fetching the CSRF token. The default is `head`.
+ - `url`: The URL for fetching the CSRF token. The default is the resource path without parameters.
+
+Global configuration `cds.env.features.fetch_csrf = true` is deprecated.
```
Comment (johannes-vogel) [inline node.js/remote-services.md:79]:
```suggestion
```

support for it is removed.

**Review by renejeglinsky — APPROVED** [review]

**Pull request merged**